internal/util/diff/diff.go

```diff
@@ -123,6 +123,11 @@ func (c *Command) Run(ctx context.Context) error {
 		return errors.Errorf("package missing Kptfile at '%s': %v", c.Path, err)
 	}
 
+	// Return early if upstream is not set
+	if kptFile.Upstream == nil || kptFile.Upstream.Git == nil {
+		return errors.Errorf("no upstream set")
```

Comment:
nit: I suggest including the package path in the error message. 

```suggestion
		return errors.Errorf("package missing upstream in Kptfile at '%s', c.Path")
```

Reply:
Good call, thanks!